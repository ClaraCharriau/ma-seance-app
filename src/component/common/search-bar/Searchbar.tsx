import { FormEvent, useState } from 'react';
import { createSearchParams, useNavigate, useSearchParams } from 'react-router-dom';
import style from './Searchbar.module.css';

const Searchbar = () => {
    const [searchBarParams, setSearchBarParams] = useSearchParams();
    const [query, setQuery] = useState<string>('');
    const navigate = useNavigate();

    const submitSearch = (e: FormEvent<HTMLFormElement>) => {
        e.preventDefault();
        searchInput();
    };

    const searchInput = () => {
        searchBarParams.set('q', query);
        setSearchBarParams(searchBarParams);
        navigate({ pathname: '/search', search: createSearchParams(searchBarParams).toString() }, { state: { query } });
    };

    return (
        <form onSubmit={event => submitSearch(event)} className={style.searchLayout}>
            <button className={style.searchBtn} type="submit">
                <label htmlFor="search">
                    <svg width="31" height="30" viewBox="0 0 31 30" fill="none" xmlns="http://www.w3.org/2000/svg">
                        <path
                            d="M14.3009 23.75C19.7895 23.75 24.2388 19.2728 24.2388 13.75C24.2388 8.22715 19.7895 3.75 14.3009 3.75C8.81238 3.75 4.36304 8.22715 4.36304 13.75C4.36304 19.2728 8.81238 23.75 14.3009 23.75Z"
                            stroke="#FAB32D"
                            strokeWidth="2"
                            strokeLinecap="round"
                            strokeLinejoin="round"
                        />
                        <path
                            d="M26.7233 26.25L21.3196 20.8125"
                            stroke="#FAB32D"
                            strokeWidth="2"
                            strokeLinecap="round"
                            strokeLinejoin="round"
                        />
                    </svg>
                </label>
            </button>
            <input
                className={style.searchInput}
                type="text"
                name="search"
                id="search"
                placeholder="Rechercher un cinÃ©ma, un film..."
                onChange={event => setQuery(event.target.value)}
            />
        </form>
    );
};

export default Searchbar;
