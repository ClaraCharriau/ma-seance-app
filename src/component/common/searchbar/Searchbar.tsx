import style from './Searchbar.module.css';

const Searchbar = () => {
    return (
        <div className={style.searchLayout}>
            <button className={style.searchBtn}>
                <svg width="31" height="30" viewBox="0 0 31 30" fill="none" xmlns="http://www.w3.org/2000/svg">
                    <path
                        d="M14.3009 23.75C19.7895 23.75 24.2388 19.2728 24.2388 13.75C24.2388 8.22715 19.7895 3.75 14.3009 3.75C8.81238 3.75 4.36304 8.22715 4.36304 13.75C4.36304 19.2728 8.81238 23.75 14.3009 23.75Z"
                        stroke="#FAB32D"
                        strokeWidth="2"
                        strokeLinecap="round"
                        strokeLinejoin="round"
                    />
                    <path
                        d="M26.7233 26.25L21.3196 20.8125"
                        stroke="#FAB32D"
                        strokeWidth="2"
                        strokeLinecap="round"
                        strokeLinejoin="round"
                    />
                </svg>
            </button>
            <input className={style.searchInput} type="text" placeholder="Rechercher un cinÃ©ma, un film..." />
        </div>
    );
};

export default Searchbar;
