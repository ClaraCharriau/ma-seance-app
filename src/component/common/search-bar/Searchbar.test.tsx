import { fireEvent, render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Searchbar from './Searchbar';

describe('Search bar component tests', () => {
    it('should render search bar', () => {
        // Given
        const { getByPlaceholderText } = render(
            <BrowserRouter>
                <Searchbar />
            </BrowserRouter>
        );

        // Then
        expect(getByPlaceholderText('Rechercher un cinéma, un film...')).toBeInTheDocument();
    });

    it('should render search field with input', () => {
        // Given
        const component = render(
            <BrowserRouter>
                <Searchbar />
            </BrowserRouter>
        );
        const searchInput = component.getByPlaceholderText('Rechercher un cinéma, un film...');

        // When
        fireEvent.change(searchInput, { target: { value: 'test' } });
        fireEvent.keyDown(searchInput, { key: 'enter', keyCode: 13 });

        // Then
        expect(component.baseElement).toMatchSnapshot();
    });
});
