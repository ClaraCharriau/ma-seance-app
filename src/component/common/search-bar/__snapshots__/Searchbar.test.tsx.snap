// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search bar component tests should render search field with input 1`] = `
<body>
  <div>
    <form
      class="searchLayout"
    >
      <button
        class="searchBtn"
        type="submit"
      >
        <label
          for="search"
        >
          <svg
            fill="none"
            height="30"
            viewBox="0 0 31 30"
            width="31"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14.3009 23.75C19.7895 23.75 24.2388 19.2728 24.2388 13.75C24.2388 8.22715 19.7895 3.75 14.3009 3.75C8.81238 3.75 4.36304 8.22715 4.36304 13.75C4.36304 19.2728 8.81238 23.75 14.3009 23.75Z"
              stroke="#FAB32D"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
            <path
              d="M26.7233 26.25L21.3196 20.8125"
              stroke="#FAB32D"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </label>
      </button>
      <input
        class="searchInput"
        id="search"
        name="search"
        placeholder="Rechercher un cinÃ©ma, un film..."
        type="text"
      />
    </form>
  </div>
</body>
`;
